### PR TITLE
Add some cases about attachInterrupt/detachInterrupt function

### PR DESCRIPTION
**Pull request by v-fional**

### CODE REVIEW

File: AZ3166/tests/UnitTest/AttchInterrupt.ino [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  attachInterrupt(PB_7, blink, CHANGE);
+  attachInterrupt(PB_13, blink, CHANGE);
+  attachInterrupt(PB_15, blink, CHANGE);
+}
+
+test(PA_5)
+{
+  for (int i = 0; i < 5; i++)
+  {
+    digitalWrite(PA_5, !digitalRead(PA_5));
```
[the remark this answers is not in the document]
Comment (v-fional) [reply AZ3166/tests/UnitTest/AttchInterrupt.ino]: Updated, as of now, we use the asserEqual() to check the LED_USER state

File: AZ3166/tests/UnitTest/AttchInterrupt.ino [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  {
+    digitalWrite(PB_15, !digitalRead(PB_15));
+    delay(1000);
+  }
+}
+
+void blink()
+{
+  state = !state;
+
```
[the remark this answers is not in the document]
Comment (v-fional) [reply AZ3166/tests/UnitTest/AttchInterrupt.ino]: Updated

File: AZ3166/tests/UnitTest/AttchInterrupt.ino [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  pinMode(PB_15, OUTPUT);
+
+  digitalWrite(PB_2, state);
+  rgb.setColor(0, 0, 0);
+
+  attachInterrupt(PA_4, blink, FALLING);
+  attachInterrupt(PA_5, blink, CHANGE);
+  attachInterrupt(PA_10, blink, FALLING);
+  attachInterrupt(PB_0, blink, CHANGE);
+  attachInterrupt(PB_2, blink, CHANGE);
```
[the remark this answers is not in the document]
Comment (v-fional) [reply AZ3166/tests/UnitTest/AttchInterrupt.ino]: Updated

File: AZ3166/tests/UnitTest/AttchInterrupt.ino (hunk 1 above)
[the remark this answers is not in the document]
Comment (ihere1) [reply AZ3166/tests/UnitTest/AttchInterrupt.ino]: Expected state should not be updated in blink. It should be calculated by loop time, initial state, and trigger method.

[the remark this answers is not in the document]
Comment (v-fional) [reply AZ3166/tests/UnitTest/AttchInterrupt.ino]: Updated, we set a variable to check attachInterrupt() now

File: AZ3166/tests/UnitTest/AttchInterrupt.ino [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  digitalWrite(pin, LOW);
+  assertEqual(counter, 1);
+  delay(LOOP_DELAY);
+
+  digitalWrite(pin, HIGH);
+  assertEqual(counter, 2);
+  delay(LOOP_DELAY);
+
+  detachInterrupt(pin);
+}
```
Comment (ihere1) [inline AZ3166/tests/UnitTest/AttchInterrupt.ino]: Add more test after detach: write digital and assert counter not changed.

**Pull request merged**